snuba/utils/streams/kafka.py

```diff
@@ -34,6 +62,9 @@ class TransportError(ConsumerError):
     pass
 
 
+ConsumerState = Enum("ConsumerState", ["CONSUMING", "ASSIGNING", "REVOKING", "CLOSED"])
```

Comment:
The only state that is really used for validation here are `ASSIGNING` (to control the behavior of `seek`) and `CLOSED` (to ensure several methods cannot be called after the consumer has been closed that do not call the Confluent API directly.)